feat(home-page): vision & mission section

nethmalds opened the pull request.

nethmalds committed:
feat: adding components and initial changes

nethmalds committed:
feat: adding components and initial changes

nethmalds committed:
feat(images): add new logos and visual assets for IEEE NSBM sections
- Added IEEE NSBM CS Logo (Color and Color with White Text)
- Added IEEE NSBM SB Logo (Blue variant)
- Added Mission and Vision SVG files
- Added WIE Logo (Color variant)

feat(ui): implement ImageSwitcher component for dynamic image display
- Created ImageSwitcher.tsx component

fix(styles): update global styles and homepage layout
- Modified globals.css for style improvements
- Updated page.tsx for layout adjustments

nethmalds committed:
fix(styles): update global styles and layout adjustments
- Modified globals.css for improved styling
- Updated page.tsx to refine layout and structure
- Adjusted ImageSwitcher.tsx component for better image handling

nethmalds committed:
feat: update logo and add VMSection component
- Replaced old IEEE NSBM SB logo with new colored version
- Updated `page.tsx` and `ImageSwitcher.tsx` to reflect new changes
- Added new `VMSection.tsx` component

nethmalds committed:
refactor: update layout and functionality in page and components
- Updated `page.tsx` for improved layout and structure
- Modified `ImageSwitcher.tsx` for enhanced image handling
- Adjusted `VMSection.tsx` for better content display

AyoItsYas commented on bun.lockb:
The `./bun.lockb` file was modified, please commit the `./package.json` file along with it. The lock file change is a result of a dependency change.

AyoItsYas commented on public/Images/IEEE NSBM CS LOGO REBUILD - COLOR 1.svg:
In modern react we do not loose reference images/assets by passing the asset location to a `src` prop. Why? because if the image location is changed/file is renamed no errors would show up in the build resulting a website with broken image links. Instead we place the images in `./src/assets` and import the images in JSX. Switch to the `release/development` branch and refer how the footer logo was placed in the code base.

AyoItsYas commented on public/Images/IEEE NSBM CS LOGO REBUILD - COLOR 1.svg:
Also rename the images so that it follows kebab case, ex: `public/Images/IEEE NSBM CS LOGO REBUILD - COLOR 1.svg` -> `public/images/ieee-nsbm-cs-logo-rebuild-color-1.svg`

AyoItsYas commented on src/app/globals.css:
The `./global.css` file as it stands for applies globally we do not want these css rules in another sub page, to avoid this we make a separate css file in the folder where the component is placed. But looking at the css included you do not need that, these css rules can simply be applied with tailwind, use tailwind directives to change colors and add a filter.

AyoItsYas commented on src/components/ImageSwitcher.tsx:
Rename the file to follow kebab case, `image-switcher.tsx`

AyoItsYas commented on src/components/VMSection.tsx:
Rename the file to follow kebab case.

AyoItsYas commented on src/components/VMSection.tsx:
Use the `Image` component provided from next.js to avoid layout shifting. https://nextjs.org/docs/pages/api-reference/components/image

AyoItsYas commented on src/components/ImageSwitcher.tsx:
```diff
@@ -0,0 +1,50 @@
+
+  return (
+    <div>
+      {theme === "light" ? (
```
The logic here can be simplified instead of de duping all the lines use a ternary operator on the prop values that change.

AyoItsYas replied:
```
<Component
  someProp={`common-value ${darkMode ? 'bg-black' : 'bg-white'}`}
/>
```

nethmalds committed:
chore(assets): renames to comply with kebab case

nethmalds committed:
chore: update global styles and rename components for consistency
- Updated `globals.css` with new styles
- Modified `page.tsx` for content/layout adjustments
- Renamed `ImageSwitcher.tsx` to `image-switcher.tsx` for consistency
- Renamed `VMSection.tsx` to `vm-section.tsx` for consistency
- Updated `tailwind.config.ts` for new configurations

nethmalds committed:
chore(husky): update pre-push hook
style(globals.css): update global styles
feat(image-switcher): enhance image switching functionality
feat(vm-section): improve VM section component
